src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -710,7 +717,16 @@ private void Serialize(string filename)
                     var folder = Path.GetDirectoryName(filename);
                     if (!Directory.Exists(folder))
                     {
-                        Directory.CreateDirectory(folder);
+                        try
+                        {
+                            Directory.CreateDirectory(folder);
+                        }
+                        catch (UnauthorizedAccessException)
+                        {
+                            // service accounts won't be able to create directory
```

Comment:
I haven't thought through the implications - but should we have a fallback?  e.g. `$env:TEMP`?